Rename most pod_id usage to pod_name in KubernetesExecutor

jedcunningham opened the pull request:
We were using pod_id in a lot of place, where really it is just the pod name. I've renamed it, where it is easy to do so, so things are easier to follow.

jedcunningham commented on airflow/executors/kubernetes_executor.py:
```diff
@@ -353,12 +353,12 @@ def run_next(self, next_job: KubernetesJobType) -> None:
 
-    def delete_pod(self, pod_id: str, namespace: str) -> None:
-        """Deletes POD."""
+    def delete_pod(self, pod_name: str, namespace: str) -> None:
```
I guess arguably this is part of the public api?

o-nikolas replied:
This is an interesting grey area. We say in the new public interface docs that executors are public and we even encourage extending them:
> Airflow has a set of Executors that are considered public. You are free to extend their functionality

But this is not the `KubernetesExecutor` class itself that you're modifying it's the job watcher singleton (which is indirectly used by the executor class via the scheduler class, which is itself passed in as a parameter to the executor).

So I could see this being a private class who's interface we don't guarantee to be consistent release to release.

I'm interested to hear what others think though.

dstandish replied:
> I'm interested to hear what others think though.

me too. i commented on the discussion thread.  ideally i think only base executor is public -- i.e. the executor interface --  and instances are not.

potiuk replied:
I commented in the thread too :) https://lists.apache.org/thread/xv5p733j2romrhsl174fpnvfb8fw30co

jedcunningham replied:
Outcome: it's not part of the public api 👍

jedcunningham commented on airflow/executors/kubernetes_executor.py:
```diff
@@ -368,17 +368,17 @@ def delete_pod(self, pod_id: str, namespace: str) -> None:
                 raise
 
-    def patch_pod_executor_done(self, *, pod_id: str, namespace: str):
+    def patch_pod_executor_done(self, *, pod_name: str, namespace: str):
```
This one isn't in a release yet, so it's fair game.

jedcunningham replied:
I actually moved this specific change out into #29147, so it can be easily cherry-picked to 2.5.2.

jedcunningham commented on airflow/executors/kubernetes_executor.py:
```diff
@@ -792,28 +793,31 @@ def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[Task
+        self,
+        kube_client: client.CoreV1Api,
+        pod: k8s.V1Pod,
+        tis_to_flush_by_key: dict[TaskInstanceKey, k8s.V1Pod],
```
Maybe this one too? Ugh, this is the worst offender, because these are really TI keys.

o-nikolas replied:
This method is in the `KubernetesExecutor` class proper so I think squarely lands in public land :grimacing:

jedcunningham replied:
We've decided that BaseExecutor is public, not all of our executors 👍

jedcunningham committed:
Rename most pod_id usage to pod_name in KubernetesExecutor

 We were using pod_id in a lot of place, where really it is just the pod
 name. I've renamed it, where it is easy to do so, so things are easier
 to follow.

potiuk reviewed (approved):
I think we are close to - or already agreed to - getting the "definition" of Public - so for me this one is "OK" to rename those.

jedcunningham commented on airflow/executors/kubernetes_executor.py:
```diff
@@ -915,8 +919,8 @@ def _adopt_completed_pods(self, kube_client: client.CoreV1Api) -> None:
                 self.log.info("Failed to adopt pod %s. Reason: %s", pod.metadata.name, e)
-            pod_id = annotations_to_key(pod.metadata.annotations)
-            self.running.add(pod_id)
+            ti_id = annotations_to_key(pod.metadata.annotations)
+            self.running.add(ti_id)
```
```suggestion
            ti_key = annotations_to_key(pod.metadata.annotations)
            self.running.add(ti_key)
```

jedcunningham committed:
Missed a new occurrence!
